EarlyWarning.cpp

```diff
@@ -0,0 +1,63 @@
+#include "EarlyWarning.h"
+#include "Localization.h"
+
+
+
+
+bool isTemperatureWarning(float temperature, bool isCelsius) {
+    float tolerance = TEMPERATURE_CRITICAL_HIGH * TEMPERATURE_WARNING_TOLERANCE;
+    if (isCelsius) {
+        tolerance = 1.5f;   // Adjust tolerance for Celsius if necessary
+    }
+    return (((temperature >= TEMPERATURE_CRITICAL_LOW) &&

```

Comment:
Explore methods to reduce the cyclomatic complexity here, by asking ChatGPT
Objective of the exercise is to explore different ways to code. If you do something here, it doesn't mean you will do that always in future.